homeassistant/components/vulcan/config_flow.py

```diff
@@ -0,0 +1,363 @@
+                errors["base"] = "unknown"
+            if errors == {}:
+                account = credentials["account"]
+                keystore = credentials["keystore"]
+                client = Vulcan(keystore, account)
+                _students = await client.get_students()
+                await client.close()
+
+                if len(_students) > 1:
+                    # pylint:disable=attribute-defined-outside-init
+                    self.account = account
+                    self.students = _students
+                    return await self.async_step_select_student()
+                _student = _students[0]
+                await self.async_set_unique_id(str(_student.pupil.id))
+                self._abort_if_unique_id_configured()
+                return self.async_create_entry(
+                    title=f"{_student.pupil.first_name} {_student.pupil.last_name}",
+                    data={
+                        "student_id": str(_student.pupil.id),
+                        "login": account.user_login,
+                    },
+                )
+
+        return self.async_show_form(
+            step_id="auth",
+            data_schema=vol.Schema(LOGIN_SCHEMA),
+            errors=errors,
+        )
+
+    async def async_step_select_student(self, user_input=None):
+        """Allow user to select student."""
+        errors = {}
+        students_list = {}
+        if self.students is not None:
+            for student in self.students:
+                students_list[
+                    str(student.pupil.id)
+                ] = f"{student.pupil.first_name} {student.pupil.last_name}"
+        if user_input is not None:
+            student_id = user_input["student"]
+            await self.async_set_unique_id(str(student_id))
+            self._abort_if_unique_id_configured()
+            return self.async_create_entry(
+                title=students_list[student_id],
+                data={
+                    "student_id": str(student_id),
+                    "login": self.account.user_login,
+                },
+            )
+
+        data_schema = {
+            vol.Required(
+                "student",
+            ): vol.In(students_list),
+        }
+        return self.async_show_form(
+            step_id="select_student",
+            data_schema=vol.Schema(data_schema),
+            errors=errors,
+        )
+
+    async def async_step_select_saved_credentials(self, user_input=None):
+        """Allow user to select saved credentials."""
+        errors = {}
+        credentials_list = {}
+        file_list = os.listdir(".vulcan")
+        for file in file_list:
+            if file.startswith("account-") and file.endswith(".json"):
+                if file.replace("account", "keystore") in file_list:
+                    credentials_list[os.path.join(".vulcan", file)] = file[
+                        len("account-") :
+                    ][: -len(".json")]
```

Comment:
Credentials need to be stored in the configuration entry, not in custom JSON files.